electron/main.ts

```diff
@@ -101,22 +104,35 @@ app.on('activate', () => {
 });
 
 function openFile() {
+  const fileChannel = 'file-content';
   dialog
     .showOpenDialog({
-      properties: ['openFile', 'multiSelections'],
-      filters: [{ name: 'Maps', extensions: ['json', 'geojson'] }]
+      properties: ['openDirectory']
     })
     .then((data) => {
       if (!data) return;
-
       const { filePaths } = data;
-      // TODO: Change the fact that implementation currently only supports opening one file at a time.
-
-      fs.readFile(filePaths[0], 'utf-8', (err, data) => {
-        // We could also probably just send an alert/dialogue saying something went wrong
-        if (err) return;
+      console.log(`filePath`, filePaths[0]);
+      let jsonFiles: any[] = [];
 
-        win?.webContents.send('file-content', JSON.parse(data.toString()));
+      readDir(filePaths[0]).then((files: any) => {
```

Comment:
Could we use `Promise` versions of the `fs` functions, and maybe look into using `async` / `await` in this function as well just to improve readability.

Reply:
There are Promise versions of the FS functions? I couldn't find anything, which is why I used the util library LOL

Reply:
Just import from `fs/promises` instead of just `fs`